test/polygon_corpus.cpp

```diff
@@ -4747,3 +4747,10 @@ TEST(Polygon, Woodgrain2) {
   });
   TestPoly(polys, 9, 0.00145);
 }
+
+TEST(Polygon, Separate_Quad_And_Tri) {
+  Polygons polys;
+  polys.push_back({{3, 1}, {3, 0}, {2, 0}});
```

Comment:
I should have noticed this earlier - these polygons are not valid (not even epsilon-valid). The triangle is CW and outside of the quad. So I guess I would say this is expected behavior, in that the returned triangulation is manifold, but not entirely CCW. 

I'm even seeing that it's producing your expected 3 triangles - only in the duplicate case (where we put a second copy right next to the first) is it producing 10 tris instead of 6, since it's attempting to fit holes into contours.

What would be your expected behavior?

Reply:
Oh, I didn't notice that at first either. OpenSCAD is way too forgiving about winding order (we use clipper to sanitize input polygons). I still got odd results in OpenSCAD, but the Manifold test now passes. I'll go back and debug where that odd behavior originates.